packages/insomnia/src/account/session.ts

```diff
@@ -195,6 +211,119 @@ async function _unsetSessionData() {
+    if ('credentials' in cred) {
+      if ('secretAccessKey' in cred.credentials) {
+        removals.push(
+          cloudCredential.update(cred, {
+            // AWS
+            credentials: { ...cred.credentials, secretAccessKey: '', sessionToken: '' },
+          }),
+        );
+        continue;
+      }
+      // hashicorp
+      if ('access_token' in cred.credentials) {
+        removals.push(cloudCredential.update(cred, { credentials: { ...cred.credentials, access_token: '' } }));
+        continue;
+      }
+      if ('client_secret' in cred.credentials) {
+        removals.push(cloudCredential.update(cred, { credentials: { ...cred.credentials, client_secret: '' } }));
+        continue;
+      }
+      if ('secret_id' in cred.credentials) {
+        removals.push(
+          cloudCredential.update(cred, { credentials: { ...cred.credentials, secret_id: '', role_id: '' } }),
+        );
+        continue;
+      }
+      // azure
+      if ('accessToken' in cred.credentials) {
+        removals.push(cloudCredential.update(cred, { credentials: { ...cred.credentials, accessToken: '' } }));
+      }
+    }
+  }
+
+  for (const backend of LLM_BACKENDS) {
+    const apiKey = await pluginData.getByKey(AI_PLUGIN_NAME, `${backend}.apiKey`);
+    if (apiKey) {
+      removals.push(pluginData.removeByKey(AI_PLUGIN_NAME, `${backend}.apiKey`));
+      if (backend === (await window.main.llm.getActiveBackend())) {
+        removals.push(window.main.llm.clearActiveBackend());
+      }
+    }
+  }
+
+  const customGitRepos = await gitRepository.all();
+  for (const repo of customGitRepos) {
+    if (!repo.credentials) continue; // unauthenticated git repositories need not be removed
+    if (isGitCredentialsOAuth(repo.credentials)) {
+      if (repo.credentials.token) {
+        removals.push(_removeGitRepository(repo));
+      }
+    } else if (repo.credentials.password) {
+      removals.push(_removeGitRepository(repo));
+    }
+  }
+
+  const proxySettings = await settings.get();
+  if (proxySettings.httpProxy?.includes('@')) {
+    removals.push(settings.update(proxySettings, { httpProxy: '' }));
+  }
+  if (proxySettings.httpsProxy?.includes('@')) {
+    removals.push(settings.update(proxySettings, { httpsProxy: '' }));
+  }
+
+  await Promise.all(removals);
+}
+
+/*
+ * Removes a git repo from the database and unlinks it from associated projects and workspaces.
+ *
+ * Clearing only the credentials from the git repo would leave the associated workspace in a broken
+ * state where the user would need to manually reset the git repo settings, and that is not
+ * immediately apparent in the UI.
+ *
+ * This way, the user can simply re-connect with their settings.
+ *
+ * Almost identical to resetGitRepoAction in main/git-service.ts, but walks through
+ * each model instance individually to clear them all out.
+ *
+ */
+async function _removeGitRepository(repo: GitRepository) {
+  const projects = await project.all();
```

Comment:
Same suggestion as the workspaceMetas:
```suggestion
  const projects = await database.find<Project>(models.project.type, {
    gitRepositoryId: repo._id
  })
```

Reply:
doneee